apps/designer/app/designer/shared/fonts-manager/fonts-manager.tsx

```diff
@@ -80,62 +92,116 @@ export const FontsManager = ({ value, onChange }: FontsManagerProps) => {
+    fontItems.find((item) => item.label === value) ??
+    fontItems.find((item) => item.type !== "category") ??
+    null;
+
+  const {
+    items: filteredItems,
+    resetFilter,
+    ...comboboxProps
+  } = useCombobox({
+    items: fontItems,
+    value: selectedItem,
+    itemToString(item) {
+      return item?.type === "category" ? "" : item?.label ?? "";
+    },
+    onItemSelect(value) {
+      if (value !== null) {
+        onChange(value.label);
+      }
+    },
+  });
+
+  const items = groupItems(filteredItems);
+
+  useEffect(resetFilter, [fontItems.length, resetFilter]);
+
+  return {
+    items,
+    openMenuItem,
+    setOpenMenuItem,
+    handleDelete: handleDeleteByLabel,
+    ...comboboxProps,
+  };
+};
+
+type FontsManagerProps = {
+  value: string;
+  onChange: (value: string) => void;
+};
+
+export const FontsManager = ({ value, onChange }: FontsManagerProps) => {
+  const {
+    items,
+    getInputProps,
+    handleDelete,
+    getComboboxProps,
+    getMenuProps,
+    getItemProps,
+    openMenuItem,
+    setOpenMenuItem,
+  } = useLogic({ value, onChange });
 
   return (
-    <Flex
-      gap="3"
-      direction="column"
-      css={{ padding: "$1", paddingTop: "$2", height: 460, overflow: "hidden" }}
-    >
-      <Box css={{ padding: "$2" }}>
+    <Flex direction="column" css={{ overflow: "hidden", py: "$1" }}>
+      <Flex css={{ py: "$2", px: "$3" }} gap="2" direction="column">
         <AssetUpload type="font" />
-      </Box>
-
+        <SearchField
+          {...getInputProps({ value: undefined })}
+          autoFocus
+          placeholder="Search"
+          onBlur={undefined}
+        />
+      </Flex>
+      <Separator css={{ my: "$1" }} />
+      {items.length === 0 && <NotFound />}
       <Flex
         {...getComboboxProps()}
-        css={{ flexDirection: "column", gap: "$3" }}
+        css={{
+          flexDirection: "column",
+          gap: "$3",
+          px: "$3",
+        }}
       >
-        <TextField
-          type="search"
-          prefix={<MagnifyingGlassIcon />}
-          {...getInputProps()}
-        />
-        <fieldset>
-          <legend>Choose an item</legend>
-          <Flex {...menuProps} css={{ flexDirection: "column" }}>
-            {items.map((item, index) => {
-              return (
-                <ComboboxListboxItem
-                  {...getItemProps({ item, index })}
-                  key={index}
+        <Listbox {...getMenuProps()}>
+          {items.map((item, index) => {
+            return (
+              <Fragment key={index}>
+                {item.type === "category" && index !== 0 && (
+                  <Separator css={{ my: "$1" }} />
+                )}
+                <ListboxItem
+                  {...getItemProps({
+                    item,
+                    index,
+                    highlighted: item === openMenuItem ? true : undefined,
+                  })}
+                  disabled={item.type === "category"}
```

Comment:
Instead of using a special "category" item, I would probably split array into two here, and render:

 - Category header
 - First array
 - Category header
 - Second array

Reply:
yep, I hit a bunch of bugs because of how I did it, will redo it now like you said